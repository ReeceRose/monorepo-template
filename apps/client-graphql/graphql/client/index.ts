import { NextPage } from 'next';
import { withUrqlClient, WithUrqlProps } from 'next-urql';

export function withGraphQL(page: NextPage<WithUrqlProps>) {
  return withUrqlClient(() => ({
    url: process.env.GRAPHQL_URL || 'http://localhost:3007/query',
    // requestPolicy: 'cache-and-network',
  }))(page);
}
