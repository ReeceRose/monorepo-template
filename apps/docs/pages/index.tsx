import { Button } from '@/shared/Components';

export default function Docs() {
  return (
    <div>
      <h1>Docs</h1>
      <Button />
    </div>
  );
}
