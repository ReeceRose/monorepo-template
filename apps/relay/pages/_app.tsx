import { AppProps } from 'next/app';
import { useEffect, useState } from 'react';
import { RelayEnvironmentProvider } from 'react-relay';

import { createEnvironment } from 'environment';

import '@/styles/globals.css';

function MyApp({ Component, pageProps }: AppProps): JSX.Element {
  useEffect(() => {
    // Note, a button is required to switch the themes and it will also need to call classList.add.
    if (localStorage.theme) {
      document.documentElement.classList.add(localStorage.theme);
    }
  }, []);

  const [env] = useState(() => {
    if (typeof window === 'undefined') {
      return createEnvironment().environment;
    }

    const recordsEl = document.getElementById('relay--records');

    const records =
      recordsEl && recordsEl.innerText && JSON.parse(recordsEl.innerText);

    return createEnvironment({ records }).environment;
  });

  return (
    <RelayEnvironmentProvider environment={env}>
      <Component {...pageProps} />
    </RelayEnvironmentProvider>
  );
}

export default MyApp;
