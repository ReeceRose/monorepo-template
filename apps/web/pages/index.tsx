import { SEO } from 'ui/components/SEO';

export default function Home(): JSX.Element {
  return (
    <div>
      <SEO
        title="CHANGE_ME"
        description="CHANGE_ME"
        siteURL={process.env.SITE_URL || 'https://localhost:3000'}
      />
    </div>
  );
}
