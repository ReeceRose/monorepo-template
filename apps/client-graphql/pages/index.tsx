import { useQuery } from 'urql';
import { withUrqlClient } from 'next-urql';

import { SEO } from 'ui/components/';
import { TodoList } from 'ui/components/Todo/';
import { GenericNavbar } from 'ui/components/Navbars/';

import { TodosQueryDocument } from '@/generated/graphql';

// eslint-disable-next-line @typescript-eslint/no-unused-vars
export default withUrqlClient((_ssrExchange, _ctx) => ({
  url: process.env.GRAPHQL_URL || 'http://localhost:3007/query',
}))(Home);

export function Home(): JSX.Element {
  const [result] = useQuery({
    query: TodosQueryDocument,
  });

  const markAsCompleted = (id: string): boolean => {
    console.log('marking as done on ', id);
    return true;
  };

  return (
    <>
      <SEO
        title="Todo list"
        description="Todo list powered by GraphQL"
        siteURL={process.env.SITE_URL || 'https://localhost:3001'}
      />
      <main>
        <GenericNavbar />
        <div className="pt-20">
          <TodoList
            data={result.data?.todos}
            markAsCompleted={markAsCompleted}
          />
        </div>
      </main>
    </>
  );
}
