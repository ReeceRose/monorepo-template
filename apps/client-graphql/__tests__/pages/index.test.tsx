import { render } from '@testing-library/react';
import { Provider } from 'urql';

import { Home } from '@/pages/index';

import { mockGraphQlClient } from '@/testing/mocks/client';

test('Index page matches snapshot', () => {
  const { asFragment } = render(
    // @ts-expect-error: Mock not fully fleshed out
    <Provider value={mockGraphQlClient}>
      <Home />{' '}
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
