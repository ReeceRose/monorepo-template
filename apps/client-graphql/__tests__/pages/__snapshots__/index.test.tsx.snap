// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <nav
      class="fixed top-0 z-50 flex flex-wrap items-center justify-between w-full px-2 py-3 bg-white shadow navbar-expand-lg"
    >
      <div
        class="container flex flex-wrap items-center justify-between px-4 mx-auto"
      >
        <div
          class="relative flex justify-between w-full lg:w-auto lg:static lg:block lg:justify-start"
        >
          <div
            class="flex items-center text-2xl font-black text-gray-800 uppercase align-middle dark:text-white"
          >
            Client - GraphQL
          </div>
        </div>
      </div>
    </nav>
    <div
      class="pt-20"
    >
      <div
        class="flex flex-col mx-auto sm:w-full md:w-8/12"
      >
        <div
          class="overflow-x-auto sm:-mx-6 lg:-mx-8"
        >
          <div
            class="inline-block min-w-full py-2 sm:px-6 lg:px-8"
          >
            <div
              class="overflow-hidden shadow-md sm:rounded-lg"
            >
              <table
                class="min-w-full"
              >
                <thead
                  class="bg-gray-50 dark:bg-gray-700"
                >
                  <tr>
                    <th
                      class="w-2/3 px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-700 uppercase dark:text-gray-400"
                      scope="col"
                    >
                      Description
                    </th>
                    <th
                      class="w-1/4 px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-700 uppercase dark:text-gray-400"
                      scope="col"
                    >
                      Completed
                    </th>
                    <th
                      class="relative px-6 py-3"
                      scope="col"
                    >
                      <span
                        class="sr-only"
                      >
                        Mark as completed
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    class="bg-white border-b dark:bg-gray-800 dark:border-gray-700"
                  >
                    <td
                      class="px-6 py-4 text-sm font-medium text-gray-900 whitespace-nowrap dark:text-white"
                    >
                      No data
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
   
</DocumentFragment>
`;
